### ISSUE TITLE
the command of readiness for type exec does not work

### ISSUE THREAD

**Issue by njuicsgz** [GoogleCloudPlatform/kubernetes#7352]
## Description
The readinessProbe does not work as httpGet. I checked the kubelet log and found Probe() function in probe.go has not been called. Am I wrong in anywhere?

```
root@paas-185:~/paas/github/kubernetes/clusters/xa/projects/test# kubectl version
Client Version: version.Info{Major:"0", Minor:"15", GitVersion:"v0.15.0", GitCommit:"831f3e60d7cd64c61a775d6c78acce1673dd8aa9", GitTreeState:"clean"}
Server Version: version.Info{Major:"0", Minor:"15", GitVersion:"v0.15.0", GitCommit:"831f3e60d7cd64c61a775d6c78acce1673dd8aa9", GitTreeState:"clean"}
```

```
vi readinessProbe.rc.json
1 {
    2    "kind":"ReplicationController",
    3    "apiVersion":"v1beta3",
    4    "metadata":{
    5       "name":"readiness",
    6       "namespace": "test-allen"
n   7    },
    8    "spec":{
    9       "replicas":1,
   10       "selector":{
   11          "name":"readiness"
   12       },
   13       "template":{
   14          "metadata":{
!  15             "namespace": "test-allen",
   16             "labels":{
   17                "name":"readiness"
   18             }
   19          },
   20          "spec":{
   21             "containers":[
   22                {
   23                   "name":"readiness",
   24                   "image":"xa.repo.ndp.com:5000/test/readiness",
   25                   "readinessProbe": {
!  26                      "initialDelaySeconds": 2,
!  27                      "exec": {
!  28                          "command": [
!  29                            "bash",
!  30                             "-c",
+  31                             "/readinessProbe.sh"  ===> the content of this shell just "echo ok".
+  32                          ]
+  33                      }
   34                   }
   35                }
   36             ]
   37          }
   38       }
   39    }
   40 }
```

But it really works when I changed "exec" to "httpGet" like this:
```
1 {
  2    "kind":"ReplicationController",
  3    "apiVersion":"v1beta3",
  4    "metadata":{
  5       "name":"readiness",
  6       "namespace": "test-allen"
  7    },
  8    "spec":{
  9       "replicas":1,
 10       "selector":{
 11          "name":"readiness"
 12       },
 13       "template":{
 14          "metadata":{
 15             "namespace": "test-allen",
 16             "labels":{
 17                "name":"readiness"
 18             }
 19          },
 20          "spec":{
 21             "containers":[
 22                {
 23                   "name":"readiness",
 24                   "image":"xa.repo.ndp.com:5000/test/readiness",
 25                   "readinessProbe": {
 26                     "httpGet": {
 27                                     "host": "172.30.10.175",
 28                                     "path": "/",
 29                                     "port": "5000"
 30                                   }
 31                   }
 32                }
 33             ]
 34          }
 35       }
 36    }
 37 }
```

**Comment by njuicsgz** [comment]
```
bug trace:
E0428 16:22:50.907280   13465 util.go:69] Recovered from panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/util/util.go:63
/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/util/util.go:54
/usr/src/go/src/runtime/asm_amd64.s:401
/usr/src/go/src/runtime/panic.go:387
/usr/src/go/src/runtime/panic.go:42
/usr/src/go/src/runtime/sigpanic_unix.go:26
/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/probe.go:232
/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/probe.go:237
/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/probe/exec/exec.go:41
<autogenerated>:1
/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/probe.go:157
/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/probe.go:145
/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/probe.go:136
/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/probe.go:90
/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/kubelet.go:1083
/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/kubelet.go:1158
/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/kubelet.go:259
/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/pod_workers.go:100
/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/pod_workers.go:109
/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/pod_workers.go:130
/usr/src/go/src/runtime/asm_amd64.s:2232
I0428 16:23:00.833819   13465 kubelet.go:1610] Periodic sync
I0428 16:23:00.834586   13465 kubelet.go:1375] Desired: []*api.Pod{(*api.Pod)(0xc20892aea0)}
```

**Comment by njuicsgz** [comment]
found the cause of "invalid memory address or nil pointer dereference", and will send a PR
thanks

**Comment by njuicsgz** [comment]
https://github.com/GoogleCloudPlatform/kubernetes/pull/7487

**Comment by njuicsgz** [comment]
can be closed since PR #7487 committed
